### ISSUE TITLE
Intended resolve/reject behavior of subscribeEvent() and unsubscribeEvent()

### ISSUE THREAD

**Issue by jak0bw** [eclipse/thingweb.node-wot#311]
When programming with node-wot I noticed the subscribeEvent() method does only resolve after the first Event was emitted by the Subscription Provider even if the subscription itself was successful (Tested longpoll and coap). Would it not be better to resolve after a successful subscription and let the provided callback Function handle the emitted Event? (Both consumer and consumedThing were produced with node-wot v7.0.1.) As a successful subscription I classified the subscription provider thing logging in his cli: 
```
HttpServer on port 9200 received 'GET /MyCoffeeMachine/events/not_enough' from [::ffff:127.0.0.1]:61483
ExposedThing 'MyCoffeeMachine' subscribes to event 'not_enough'
```
Is this the intended behavior for technical reasons, maybe the subscriber itself does not get a notification for a successful subscription, or does this behavior have other reasons? If this is the desired behavior for technical reasons is there a way to differentiate between a successful subscription and just no event happening versus an unsuccessful subscription? (I noticed an error message for longpoll if the subscription provider thing was never started (offline) but for coap the behavior of these two cases was identical in my tests. E. g. nothing happend until my own timout was triggered… (does subscribeEvent() have an internal timout I just did not hit?))

I did not test this to the same extent but I think that unsubscribeEvent() has a similar problem where it resolves even if the Event was never subscribed to (due to the subscription provider thing being offline).
 
A more general question in this direction is if there exits a development documentation anywhere that documents such behavior in detail (e. g. when exactly subscribeEvent() / unsubscribeEvent() resolves/rejects with which (error-)messages....).

**Comment by relu91** [comment]
I agree with you the subscription function should resolve right after the subscription is successful. I mentioned this change in this comment https://github.com/eclipse/thingweb.node-wot/pull/232#issuecomment-638042931 (see the last three lines). However, it requires a change in the API of protocol clients and we put this change on hold. 

> A more general question in this direction is if there exits a development documentation anywhere that documents such behavior in detail (e. g. when exactly subscribeEvent() / unsubscribeEvent() resolves/rejects with which (error-)messages....).

Node-wot follows the Web of Things Scripting API [specification](https://www.w3.org/TR/wot-scripting-api/). There you can find more documentation about functions and data models. For example, you can clearly spot that node-wot is not following the spec of the `subscribeEvent` method because the method should resolve as soon as the subscription request is successful (as you are saying). So I'd say that this issue is a bug and it needs to be addressed. Thanks for taking your time testing this 👍

**Comment by relu91** [comment]
Fixed in #526
